### PR TITLE
Rework `validateCAARecords` to handle multiple CAA records

### LINKED ISSUE #12: Incorrect 'CAA record is not "letsencrypt.org"'
Getting this issue for the domain `appwise.be`. Doesn't seem to be related to #11.

The CAA records are there, there are even multiple (one for wildcards, and also one for `iodef` (reporting issues). 

The call to `dns.google/resolve` results in:
```json
{
  "Answer":
  [
    {
      "name": "appwise.be.",
      "type": 257,
      "TTL": 21600,
      "data": "0 issuewild \"letsencrypt.org\""
    },
    {
      "name": "appwise.be.",
      "type": 257,
      "TTL": 21600,
      "data": "0 issue \"letsencrypt.org\""
    },
    {
      "name": "appwise.be.",
      "type": 257,
      "TTL": 21600,
      "data": "0 iodef \"mailto:caapolicy@appwise.be\""
    }
  ]
}
```

I think the code is tripping up because of this, because it assumes there should only be one `257` record with the expected CAA contents. It should instead check in the list of `257` records if any is for `issue`, and if so, it must contain the value `letsencrypt.org`.

The spec for CAA records allows the `flag` field to be something else than `0`, although I've yet to see something besides the value `0`.

----

Potential solution would be changing the [`for` loop here](https://github.com/willnode/forward-domain/blob/main/src/util.js#L124-L132) to something like:
```js
const issueRecord = data.Answer.find( item =>
  item.type == 257 && typeof item.data === 'string' && item.data.startsWith('0 issue ')
);

if (issueRecord.data !== "0 issue \"letsencrypt.org\"") {
  return issueRecord.data;
} else {
  return null;
}
```

That last `if` test could even be a simple `issueRecord.data.includes('letsencrypt.org')`. I think the quotes (`"`) are optional around a CAA `value`.

Note: I don't usually write JS code, so please ignore my formatting if it looks weird 😅

### PR DESCRIPTION
**Pull request by djbe**
Fixes #12

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -114,23 +114,24 @@ const parseTxtRecordData = (value) => {
...
-        if (typeof caaData === 'string' && caaData !== "0 issue \"letsencrypt.org\"") {
-            return caaData;
-        }
+
+    const issueRecord = data.Answer.filter(x =>
+        x.type == 257 && typeof x.data === 'string' && x.data.startsWith('0 issue ')
+    ).map(x => x.data);
+
+    // check if any record allows Let'sEncrypt (or no record at all)
+    if (issueRecords.length == 0 || issueRecords.some(x => x.includes('letsencrypt.org'))) {
```
Comment (willnode) [inline src/util.js:130]:
```suggestion
    if (issueRecords.length == 0 || issueRecords.some(x => x === "0 issue \"letsencrypt.org\"")) {
```
faster to compute

**Review by willnode — CHANGES_REQUESTED** [review]
changes requested

**Comment by willnode** [comment]
Additional context about `issuewild` vs `issue`: We need to match `issue` since this doesn't request a wildcard certificate. Maybe this need to be incorporated into error message.

File: src/util.js (hunk 1 above)
Comment (djbe) [reply src/util.js]: I mentioned this in the original issue: according to the CAA spec, the quotes are optional (in this case).

I've adapted the check to use a regex if that's allright. It'll correctly handle a case with or without quotes.

**Review by willnode — APPROVED** [review]

**Comment by djbe** [comment]
Updated the call to the validate function, to use the result and show it to the user.

**Pull request merged**